src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -93,7 +93,7 @@ private async Task AcquireTokenInternalAsync(IDictionary<string, string> brokerP
                 _logger.Verbose("User is specified for silent token request. Starting silent broker request");
                 string silentResult = _brokerHelper.GetBrokerAuthTokenSilently(brokerPayload, _activity);
                 _androidBrokerTokenResponse = CreateMsalTokenResponseFromResult(silentResult);
-                _readyForResponse.Release();
```

Comment:
This may fix the immediate problem, but the architectural issue remains - why do we end up calling broker code when the broker isn't active i.e. `.WithBroker(false)` ?

Reply:
yes...agreed and i think separating out things is a good idea. like i mentioned earlier, this is a temporary fix and we should consider something more long term. but that can happen now. wasn't sure on the urgency of this.

Reply:
agreed...we are doing it automatically in the `SetAuthenticationContinuatoinEventArgs`. will look into a fix there as well.